common/src/main/java/io/netty/util/internal/PlatformDependent.java

```diff
@@ -899,6 +905,90 @@ public static ClassLoader getSystemClassLoader() {
         }
     }
 
+    /**
+     * Return the size of a cache line for the running platform or {@code -1} if is unknown
+     * or could not be detected.
+     */
+    public static int cacheLineSize() {
+        return CACHE_LINE_SIZE;
+    }
+
+    private static int cacheLineSize0() {
+        return AccessController.doPrivileged(new PrivilegedAction<Integer>() {
+            @Override
+            public Integer run() {
+                if (IS_LINUX) {
+                    BufferedReader in = null;
+                    try {
+                        in = new BufferedReader(new InputStreamReader(new FileInputStream("/proc/cpuinfo")));
+                        String line;
+                        while ((line = in.readLine()) != null) {
+                            if (line.startsWith("cache_alignment")) {
+                                String[] parts = line.split(":");
+                                return Integer.parseInt(parts[1].trim());
+                            }
+                        }
+                    } catch (Exception cause) {
+                        logger.debug("Failed to retrieve cache line size", cause);
+                        return -1;
+                    } finally {
+                        if (in != null) {
+                            try {
+                                in.close();
+                            } catch (IOException ignore) {
+                                // ignore on close
+                            }
+                        }
+                    }
+                } else if (IS_OSX) {
+                    Process p = null;
+                    BufferedReader in = null;
+                    try {
+                        p = Runtime.getRuntime().exec(new String[] { "sysctl", "hw.cachelinesize" });
```

Comment:
IMHO common should not have this dependency... maybe its ok to just only do it for linux for now. Or just not do it at all.